Move templates from kubernetes/deployment-manager to the registry

jackgr opened the pull request.

bacongobbler commented:
Note that users still need to run a few commands as per https://github.com/deis/riak#usage, but if HA isn't a concern then I can strip away everything but the pod manifest if that makes sense.

gabrtv commented:
@bacongobbler let's change the path from `riak` to `riak-cluster` for convention.

bacongobbler commented:
done :)

gabrtv commented on riak-cluster/manifests/riak-discovery-service.yaml:
```diff
@@ -0,0 +1,31 @@
+      name: handoff
+      protocol: TCP
+  selector:
+    name: riak
```
Let's change this to `provider=riak` per the labels guidance in https://github.com/deis/helm/blob/master/docs/labels.md#provider-label

gabrtv commented on riak-cluster/manifests/riak-rc.yaml:
```diff
@@ -0,0 +1,29 @@
+  template:
+    metadata:
+      labels:
+        name: riak
```
`provider: riak`

bacongobbler commented:
hold off on merging this one; I'm going to provide both worlds:

 - HA Riak (current work)
 - standalone Riak pod (same as other examples)

gabrtv commented on riak-cluster/manifests/riak-service.yaml:
```diff
@@ -0,0 +1,18 @@
+      name: protobuf
+      protocol: TCP
+  selector:
+    name: riak
```
`provider: riak`

technosophos commented:
Dude, you are seriously ripping it up with these! Good example of using the `provider` label.

bacongobbler commented:
I dropped the clustered riak chart for now while trying to debug what's wrong with it so we can get this one out the door.

mboersma commented:
Tested with helm 0.1.0, everything looks good and I can curl the server 8098 endpoint inside the cluster. `kubectl logs riak` doesn't show anything, but maybe it wouldn't? Looks good to me.

bacongobbler commented:
Yeah that's something I need to fix up at some point. By default riak logs to a file inside the container. deis/riak was inspired by https://github.com/hectcastro/docker-riak which logs to /var/log/riak. I just need to tweak a few things inside https://github.com/deis/riak/blob/master/rootfs/bin/boot to start riak with logs in the foreground, though most of my attempts to do so have failed since I have to background the process.

One way is to use the same technique we use in deis/router but I'd rather do something more native instead of a goroutine piping logs to stdout/stderr.

Commit:
Code changes to support terminology rationalization.

LeendersR committed:
Adds Wordpress example

LeendersR committed:
Updates type references

zmerlynn committed:
Spark: Add DM template for Spark

This is a first-cut template based on
https://github.com/kubernetes/kubernetes/tree/master/examples/spark
with minor templating. As I iterate on that example to add PDs, etc.
(which will add much more variance), I'll iterate here, too.

Commit:
Move templates from deployment-manager repository.
